File: Searching/JavaCodes/Leetcode1482.java
class Solution {
   public static boolean canIMakeBouquet(int[] bloomDay,int m,int k,int day){
        int bouquets=0,flowers=0;
        for (int i = 0; i < bloomDay.length; i++) {
            if(bloomDay[i]<=day){
                flowers++;
            }
            else{
                flowers=0;
            }
            if(flowers==k){
                bouquets++;
                flowers=0;
            }
            if(bouquets==m){
                return true;
            }
        }
        return false;
    }
    public static int minDays(int[] bloomDay, int m, int k) {
        if((long)m*k> bloomDay.length){
            return -1;
        }
        int max=0;
        for (int i = 0; i < bloomDay.length; i++) {
            max=Math.max(max,bloomDay[i]);
        }
        int start=1,end=max;

        while(start<=end){
            int day=start+(end-start)/2;
            if(canIMakeBouquet(bloomDay,m,k,day)){
                end=day-1;
            }
            else{
                start=day+1;
            }
        }
        return start;
    }
}

File: LogicOverSyntaxCodes/Searching/PythonCodes/Leetcode1482.py
class Solution(object):
    def canIMakeBouquet(self, bloomDay, m, k, day):
        bouquets = 0
        flowers = 0
        
        for bloom in bloomDay:
            if bloom <= day:
                flowers += 1
            else:
                flowers = 0
            
            if flowers == k:
                bouquets += 1
                flowers = 0
            
            if bouquets == m:
                return True
        
        return False
    
    def minDays(self, bloomDay, m, k):
        if m * k > len(bloomDay):
            return -1
        
        start, end = 1, max(bloomDay)
        
        while start <= end:
            day = start + (end - start) // 2
            if self.canIMakeBouquet(bloomDay, m, k, day):
                end = day - 1
            else:
                start = day + 1
        
        return start